x-pack/legacy/plugins/actions/server/builtin_action_types/es_index.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf, Type } from '@kbn/config-schema';
+
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+
+function nullableType<V>(type: Type<V>) {
+  return schema.oneOf([type, schema.literal(null)], { defaultValue: () => null });
+}
+
+// config definition
+
+const unencryptedConfigProperties: string[] = ['index'];
+
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const ConfigSchema = schema.object({
+  index: nullableType(schema.string()),
+});
+
+// params definition
+
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+
+const IndexSingleBodySchema = schema.recordOf(schema.string(), schema.any());
+const IndexArrayBodySchema = schema.arrayOf(IndexSingleBodySchema);
+const IndexBodySchema = schema.oneOf([IndexSingleBodySchema, IndexArrayBodySchema]);
+
+type IndexArrayBodySchemaType = TypeOf<typeof IndexArrayBodySchema>;
+
+// see: https://www.elastic.co/guide/en/elastic-stack-overview/current/actions-index.html
+// - timeout not added here, as this seems to be a generic thing we want to do
```

Comment:
wondering if we _do_ want to add timeout here, as you might want to have that value be different from whatever would be set globally.

Reply:
I have a card open for this - https://github.com/elastic/kibana/projects/26#card-24087404

Yeah, I think we will probably want a global timeout, and then allow an action type to override.  Then the next question is, how can a user change these?  env var or kibana yaml config or ...

We could get more elaborate, allow timeout per action, but I don't think we need to go there just yet ...